rfcs/43-many-worlds.md

```diff
@@ -0,0 +1,522 @@
+fn transport_between_planets_system(events: EventReader<BeamMeUp>, app_commands: AppCommands, current_world: WorldLabel){
+   for event in events.iter(){
+      app_commands.move_entities(event.entities, current_world, event.target_world);
+   }
+}
+
+/// Change which world entities are being fetched from for pipelined rendering and audio
+// This system lives in LogicPlugin, and is replicated across each planet's worlds
+// and is triggered when the player changes which world they want to watch via the UI
+fn swap_planet_view(events: Events<SwapToWorld>, mut app_commands: AppCommands, current_world: WorldLabel){
+   if events.iter().next().is_some(){
+      // Demonstrates a hypothetical integration with pipelined-rendering
+      // (and eventually audio), where we can swap which world the entities
+      // to visualize and "audioize" are coming from
+      app_commands.set_source_world(current_world);
+   }
+}
+```
+
+### Advanced example: world-model AI
+
+In this example, we're using a multiple worlds approach in order to allow complex agents to simulate the world in order to plan ahead.
+By taking this strategy, we can ensure that the agents always have an accurate model of the rules of the game without code duplication.
+
+```rust
+use bevy::prelude::*;
+
+#[derive(WorldLabel)]
+struct GameRulesWorld;
+
+#[derive(WorldLabel)]
+struct SimulationWorld(Uuid);
+
+fn main(){
+   App::new()
+   // By storing all of our logic in its own world, we can quickly duplicate it
+   .spawn_world(GameRulesWorld)
+   .set_world(GameRulesWorld)
+   .add_plugin(GameLogicPlugin)
+   // We don't need or want all of the player-facing systems in our simulations
+   .set_world(CoreWorld::Main)
+   .add_plugins(DefaultPlugins)
+   .add_plugin(UiPlugin)
+   .add_plugin(RenderingPlugin)
+   .add_plugin(InputPlugin)
+   // We're sticking to a one-deep search for sanity here
+   .add_system(simulate_action_system)
+   .add_system(choose_action_system)
+   .run();
+}
+
+// This system is contained within `GameLogicPlugin`
+fn simulate_action_system(potential_actions: Res<PotentialActions>, mut app_commands: AppCommands, sim_query: Query<((), With<Simulate>)>){
+   // Create a new world for each potential action
+   for proposed_action in potential_actions.iter() {
+      // These identifiers are uniquely generated and ephemeral
+      let sim_world = SimulationWorld::new();
+
+      // Making a simulation world that's identical to the main world in all relevant ways
+      app_commands
+         .spawn_world(sim_world)
+         .clone_schedule(GameRulesWorld, sim_world)
+         // As above, blocked by bevy #1515
+         .clone_entities_to_world(sim_query.entities(), sim_world)
+         // This system will evaluate the world state, send it back with an event reporting the values
+         .add_system(evaluate_action_system.to_world(sim_world))
+         .send_event(proposed_action, sim_world);
```

Comment:
Currently, we use `EventWriter` to write events, here we can also you app_commands to send events, what will be the difference?

Also, it's possible that one system in the world wants to broadcast the event to all world since it does not know which world should respond to it. How this situation should be handled?